### ISSUE TITLE
Initialization failure UX is misleading

### ISSUE THREAD

**Issue by hausdorff** [pulumi/pulumi#3091]
Kubernetes resources can be created, but fail to initialize. Pulumi represents such failures with a set of protocol constructs that we call "initialization errors."

Resources in this state need to be fixed by the user—directly or indirectly, we expect that when an initialization failure occurs, a user will want to fix it ASAP, typically changing it until it's working correctly, continuously running `pulumi up`.

For this reason, when an initialization error is present, every subsequent run of `pulumi up` will attempt to wait for the resource to complete initializing. For example, if a `Deployment` failed to initialize because a `ConfigMap` it depends on does not exist, a subsequent run where a user adds a `ConfigMap` needs to await the `Deployment` again, even if no changes were made—this time, it should successfully initialize.

This is one example that illustrates why this model of "initialization failures" is critical to our model of (at least) Kubernetes resources.

Yet, the UX can be very confusing:

* Until the error is fixed, every `preview` will render a resource with an initialization failure is presented as a resource with an `update` pending; yet, it has a 0-line diff. No mention is made that the resource is already failing, and that this is the real reason we scheduled this update.
* Until the error is fixed, every `up` will wait for the resource to initialize correctly. To a user who doesn't know about any of the above, it will look like every `pulumi up` times out for no reason.

I wanted to register as a thing we need to fix in Q3. cc @pgavlin @lukehoban @lblackstone 

Thoughts?

**Comment by lblackstone** [comment]
I agree that this is confusing. Maybe it would be helpful to add a `status` field in the CLI for resources that previously failed and have an empty diff?

**Comment by lukehoban** [comment]
@lblackstone @hausdorff Can you guys summarize the concrete plan for a change we should make in the near term to address this?

**Comment by lblackstone** [comment]
I'd propose adding a `retry` (or similar name) status field to the CLI for resources that failed to become ready in the previous update and have an empty diff. 

This is a common case for k8s resources where the problem is actually in a dependent resource, or when it timed out waiting on ready. 

https://github.com/pulumi/pulumi-kubernetes/issues/571 is an example where the Ingress failed to be ready because of a misconfigured Service. 

https://github.com/pulumi/pulumi-kubernetes/issues/712 is an example of a resource that eventually became ready, but timed out on the first update.

**Comment by hausdorff** [comment]
From https://github.com/pulumi/pulumi/pull/3117#issuecomment-523594921:

> @lukehoban @pgavlin after experimenting with it both this way and your way, I think using the ephemeral channel is not going to be a great experience—the resources that _can_ have initialization failures use ephemeral messages prolifically, and anything you write there is immediately overwritten by the resource's status.
> 
> We could maybe bake in a `[retrying]` prefix into those messages, but I'm not sure how much better that will be.

After some thought, I'm somewhat sure we can't use ephemeral status for these messages. Thus, I favor doing something along the lines of that PR.

Unfortunately, I don't think adding a new kind of step fits into the 1.0 picture. So my recommendation is to do it the week after 1.0 launches.
